### PR TITLE
Prefix all sprintf() calls

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 7.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Issues        | N/A
| License       | MIT

see https://tideways.com/profiler/blog/new-in-php-8-4-engine-optimization-of-sprintf-to-string-interpolation

In PHP 8.4, some calls to `sprintf()` will be optimized by the compiler. I therefore propose to change our code style and force calls to that function to be prefixed with a `\`. This will tell the compiler that this function must not be looked up in the current namespace first and allow the compiler to optimize the call.

The changes have been automated with PHP CS Fixer.

### CODE REVIEW

**Comment by derrabus** [comment]
cc @beberlei

File: .php-cs-fixer.dist.php [hunk 1]
```diff
@@ -30,6 +30,7 @@
         '@Symfony:risky' => true,
         'protected_to_private' => false,
         'header_comment' => ['header' => $fileHeaderComment],
+        'native_function_invocation' => ['include' => ['@compiler_optimized', 'sprintf'], 'scope' => 'namespaced', 'strict' => true],
```
Comment (stof) [inline .php-cs-fixer.dist.php:34]: I suggest adding a TODO to remove that line again once we use PHP-CS-Fixer 3.60+, as `@compiler_optimized` will include it

**Review by OskarStark — APPROVED** [review]

**Comment by MatTheCat** [comment]
Should `.git-blame-ignore-revs` be updated?

**Comment by derrabus** [comment]
> Should `.git-blame-ignore-revs` be updated?

Will do, as soon as this PR is merged and the commit hash is final.

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @derrabus.

**Comment by derrabus** [comment]
> > Should `.git-blame-ignore-revs` be updated?
> 
> Will do, as soon as this PR is merged and the commit hash is final.

#57472

**Pull request merged**